noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -70,42 +73,67 @@ impl ReservedRegisters {
         MemoryAddress::from(ReservedRegisters::PreviousStackPointer as usize)
     }
 
-    /// Returns a user defined (non-reserved) register index.
-    fn user_register_index(index: usize) -> MemoryAddress {
-        MemoryAddress::from(index + ReservedRegisters::len())
+    /// Returns the usize one register. This will be used to perform arithmetic operations.
+    pub(crate) fn usize_one() -> MemoryAddress {
+        MemoryAddress::from(ReservedRegisters::UsizeOne as usize)
     }
```

Comment:
I've noticed a couple of places where we'd benefit from a `usize_zero` reserved register as well (functions with lots of multiplications, etc)

Reply:
Ah this isn't a pure `usize` as we'd need one for each bitsize which the brillig VM can address. Would still be a saving in general though (at least for commonly used bit sizes.)

Reply:
Yeah for the case of zero it's not used as much as usize_one in one specific bitsize, the only downside with having these reserved registers with commonly used values is that they have to be initialized in the entrypoint, presenting a constant cost for programs that don't use them. Maybe if the linker could collect which ones are actually used and only initialize those...

Reply:
Yup, it'd be pretty good, we could always reserve the registers and initialize lazily. It should help!

Reply:
The downside of doing this kind of optimization work in the noir repo is that it's a pain to get bytecode size metrics there for the aztec contracts

Reply:
Soon we will have a gates diff like we do for ACIR https://github.com/noir-lang/noir/pull/5747#issuecomment-2296667249 that we can utilize for testing specific cases. This isn't the aztec contracts, but we can then see the improvements on a sync.